htlcswitch/link.go

```diff
@@ -1590,7 +1590,16 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 			return
 		}
 
-		// TODO(roasbeef): pipeline to switch
+		settlePacket := &htlcPacket{
+			outgoingChanID: l.ShortChanID(),
+			outgoingHTLCID: idx,
+			htlc: &lnwire.UpdateFulfillHTLC{
+				PaymentPreimage: pre,
+			},
+		}
+
+		// Pipeline this settle, send it to the switch.
+		go l.forwardBatch(settlePacket)
```

Comment:
might suggest reordering this to pipeline after adding the preimage to the list of uncommitted preimages, that way if anything panics we will still write the preimage

Reply:
Done